caffe2/setup.py

```diff
@@ -20,8 +20,8 @@
 import sys
 from textwrap import dedent
 
-TOP_DIR = os.path.realpath(os.path.dirname(__file__))
-SRC_DIR = os.path.join(TOP_DIR, 'caffe2')
+SRC_DIR = os.path.realpath(os.path.dirname(__file__))
+TOP_DIR = os.path.join(SRC_DIR, '..')
```

Comment:
1. '..' is not truly cross-platform, there `os.pardir`
2. We should resolve the realpath instead of leaving `..` in it.